lib/crypto.js

```diff
@@ -696,6 +697,71 @@ exports.setEngine = function setEngine(id, flags) {
   return binding.setEngine(id, flags);
 };
 
+const kMaxUint32 = Math.pow(2, 32) - 1;
+
+function randomFillSync(buf, offset = 0, size) {
+  if (!isUint8Array(buf)) {
+    throw new TypeError('"buf" argument must be a Buffer or Uint8Array');
+  }
+
+  if (typeof offset !== 'number' || offset < 0 || offset > kMaxUint32) {
+    throw new TypeError('offset must be a number >= 0');
+  }
+
+  if (offset > kBufferMaxLength || offset > buf.length) {
+    throw new RangeError('offset out of range');
+  }
+
+  if (size === undefined) size = buf.length - offset;
+
+  if (typeof size !== 'number' || size < 0 || size > kMaxUint32) {
+    throw new TypeError('size must be a number >= 0');
+  }
+
+  if (size + offset > buf.length || size > kBufferMaxLength) {
+    throw new RangeError('buffer too small');
+  }
+
+  return binding.randomFill(buf, offset, size);
+}
+exports.randomFillSync = randomFillSync;
+
+function randomFill(buf, offset, size, cb) {
+  if (!isUint8Array(buf)) {
+    throw new TypeError('"buf" argument must be a Buffer or Uint8Array');
+  }
+
+  if (typeof offset === 'function') {
+    cb = offset;
+    offset = 0;
+    size = buf.length;
+  } else if (typeof size === 'function') {
+    cb = size;
+    size = buf.length - offset;
+  } else if (typeof cb !== 'function') {
+    throw new TypeError('"cb" argument must be a function');
+  }
+
+  if (typeof offset !== 'number' || offset < 0 || offset > kMaxUint32) {
+    throw new TypeError('offset must be a number >= 0');
+  }
+
+  if (offset > kBufferMaxLength || offset > buf.length) {
+    throw new RangeError('offset out of range');
+  }
+
+  if (typeof size !== 'number' || size < 0 || size > kMaxUint32) {
+    throw new TypeError('size must be a number >= 0');
+  }
+
+  if (size + offset > buf.length || size > kBufferMaxLength) {
+    throw new RangeError('buffer too small');
+  }
```

Comment:
I ended up moving them into two separate functions since the sync version automatically determines size if size is undefined. I'm not sure it makes sense to have the same logic in the async version though since the callback for it will be the last argument. Should I make `crypto.randomFill(buf, 0, undefined, callback)` work the same way that `crypto.randomFillSync(buf, 0)` and `crypto.randomFillSync(buf, 0, undefined)` work?